tests/smart_listener_test.py

```diff
@@ -0,0 +1,78 @@
+from tests.base import HazelcastTestCase
+from tests.util import configure_logging, random_string, event_collector
+from hazelcast.config import ClientConfig
+from time import sleep
+
+
+class SmartListenerTest(HazelcastTestCase):
+    @classmethod
+    def setUpClass(cls):
+        configure_logging()
+        cls.rc = cls.create_rc()
+        cls.cluster = cls.create_cluster(cls.rc, None)  # Default config
+        cls.m1 = cls.cluster.start_member()
+        cls.m2 = cls.cluster.start_member()
+        cls.m3 = cls.cluster.start_member()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.rc.exit()
+
+    def setUp(self):
+        client_config = ClientConfig()
+        client_config.network_config.smart_routing = True
+        self.client = self.create_client(client_config)
+        self.collector = event_collector()
+
+    def tearDown(self):
+        self.shutdown_all_clients()
+
+    # -------------------------- test_local_only ----------------------- #
+    def test_list_smart_listener_local_only(self):
+        list = self.client.get_list(random_string()).blocking()
+        list.add_listener(item_added_func=self.collector)
+        list.add('item-value')
+        sleep(5)
+        self.assertEqual(1, len(self.collector.events))
```

Comment:
better to use assertTrueEventually, then we can remove sleep call and test may finish earlier.

Reply:
I purposefully avoided using assertTrueEventually in this case, because I want to make sure that I receive exactly 1 event and not more than that. I think assertTrueEventually lets the test pass as soon as you receive that 1 event, but I want to wait to see I don't receive the event from all of the listeners to make sure smart listener works local only.